src/dist/replication/test/replica_test/unit_test/replica_test.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include <gtest/gtest.h>
+
+#include <dsn/utility/fail_point.h>
+#include "replica_test_base.h"
+
+namespace dsn {
+namespace replication {
+
+class replica_test : public replica_test_base
+{
+public:
+    dsn::app_info _app_info;
+    dsn::gpid pid = gpid(2, 1);
+
+public:
+    void SetUp() override
+    {
+        stub->install_perf_counters();
+        mock_app_info();
+        stub->generate_replica(_app_info, pid, partition_status::PS_PRIMARY, 1);
+    }
+
+    int get_write_size_exceed_threshold_count()
+    {
+        return stub->_counter_recent_write_size_exceed_threshold_count->get_value();
+    }
+
+    void mock_app_info()
+    {
+        _app_info.app_id = 2;
+        _app_info.app_name = "replica_test";
+        _app_info.app_type = "replica";
+        _app_info.is_stateful = true;
+        _app_info.max_replica_count = 3;
+        _app_info.partition_count = 8;
+    }
+};
+
+TEST_F(replica_test, write_size_limited)
+{
+    int count = 100;
```

Comment:
I think use loop is better than count

Reply:
It‘s corresponding to `count` of `write_size_exceed_threshold_count()` , and it actually it is inconsequential.

Reply:
OK, I got it. And you can write these code like this:
```
int count = 100;
for (int i =0; i < count; i++) {
        stub->on_client_write(pid, write_request);
 }
ASSERT_EQ(get_write_size_exceed_threshold_count(), count);
```